sources/api/pluto/src/main.rs

```diff
@@ -346,14 +346,26 @@ async fn generate_provider_id(
     Ok(())
 }
 
-async fn generate_private_dns_name(
-    client: &mut ImdsClient,
-    aws_k8s_info: &mut AwsK8sInfo,
-) -> Result<()> {
-    if aws_k8s_info.hostname_override.is_some() || NO_HOSTNAME_VARIANTS.contains(&aws_k8s_info.variant_id.as_str()) {
+/// generate_node_name sets the hostname_override, if it is not already specified
+async fn generate_node_name(client: &mut ImdsClient, aws_k8s_info: &mut AwsK8sInfo) -> Result<()> {
+    // hostname override provided, so we do nothing
+    if aws_k8s_info.hostname_override.is_some() {
         return Ok(());
     }
 
+    // no override source provided, and this is one of the no hostname variants
+    if aws_k8s_info.hostname_override_source.is_none()
+        && NO_HOSTNAME_VARIANTS.contains(&aws_k8s_info.variant_id.as_str())
+    {
+        return Ok(());
+    }
```

Comment:
We could get rid of `NO_HOSTNAME_VARIANTS` entirely, and just rely on `hostname-override-source` being left unset, if you make a slight tweak to https://github.com/bottlerocket-os/bottlerocket/pull/4118 and also change the [kubernetes-aws-external-cloud-provider.toml](https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/shared-defaults/kubernetes-aws-external-cloud-provider.toml) defaults to be:
```
[settings.kubernetes]
cloud-provider = "external"
hostname-override-source = "private-dns-name"
```

You've already added the migration so this would be an easy 1-line change in that PR.

That particular file is the only difference between defaults settings for `aws-k8s-1.25` and `aws-k8s-1.26`, per:
```
git diff --no-index sources/settings-defaults/aws-k8s-1.2{5,6}
```